src/test_utils/routes_variadic.rs

```diff
@@ -0,0 +1,97 @@
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use tide::Route;
+
+#[derive(Debug)]
+pub struct VariadicRoutes<State, RoutesFn>
+where
+    State: Send + Sync + 'static,
+    RoutesFn: Fn(Route<'_, Arc<State>>),
+{
+    _phantom_state: PhantomData<State>,
+    pub routes: Vec<RoutesFn>,
+}
+
+impl<State, RoutesFn> From<RoutesFn> for VariadicRoutes<State, RoutesFn>
+where
+    State: Send + Sync + 'static,
+    RoutesFn: Fn(Route<'_, Arc<State>>),
+{
+    fn from(routes: RoutesFn) -> Self {
+        VariadicRoutes {
+            _phantom_state: PhantomData,
+            routes: vec![routes],
+        }
+    }
+}
+
+impl<State, RoutesFn> From<(RoutesFn, RoutesFn)> for VariadicRoutes<State, RoutesFn>
+where
+    State: Send + Sync + 'static,
+    RoutesFn: Fn(Route<'_, Arc<State>>),
+{
+    fn from(routes: (RoutesFn, RoutesFn)) -> Self {
+        VariadicRoutes {
+            _phantom_state: PhantomData,
+            routes: vec![routes.0, routes.1],
+        }
+    }
+}
+
+impl<State, RoutesFn> From<(RoutesFn, RoutesFn, RoutesFn)> for VariadicRoutes<State, RoutesFn>
+where
+    State: Send + Sync + 'static,
+    RoutesFn: Fn(Route<'_, Arc<State>>),
+{
+    fn from(routes: (RoutesFn, RoutesFn, RoutesFn)) -> Self {
+        VariadicRoutes {
+            _phantom_state: PhantomData,
+            routes: vec![routes.0, routes.1, routes.2],
+        }
+    }
+}
+
+impl<State, RoutesFn> From<(RoutesFn, RoutesFn, RoutesFn, RoutesFn)>
+    for VariadicRoutes<State, RoutesFn>
+where
+    State: Send + Sync + 'static,
+    RoutesFn: Fn(Route<'_, Arc<State>>),
+{
+    fn from(routes: (RoutesFn, RoutesFn, RoutesFn, RoutesFn)) -> Self {
+        VariadicRoutes {
+            _phantom_state: PhantomData,
+            routes: vec![routes.0, routes.1, routes.2, routes.3],
+        }
+    }
+}
+
+impl<State, RoutesFn> From<(RoutesFn, RoutesFn, RoutesFn, RoutesFn, RoutesFn)>
+    for VariadicRoutes<State, RoutesFn>
+where
+    State: Send + Sync + 'static,
+    RoutesFn: Fn(Route<'_, Arc<State>>),
+{
+    fn from(routes: (RoutesFn, RoutesFn, RoutesFn, RoutesFn, RoutesFn)) -> Self {
+        VariadicRoutes {
+            _phantom_state: PhantomData,
+            routes: vec![routes.0, routes.1, routes.2, routes.3, routes.4],
+        }
+    }
+}
+
+impl<State, RoutesFn> From<(RoutesFn, RoutesFn, RoutesFn, RoutesFn, RoutesFn, RoutesFn)>
+    for VariadicRoutes<State, RoutesFn>
+where
+    State: Send + Sync + 'static,
+    RoutesFn: Fn(Route<'_, Arc<State>>),
+{
+    fn from(routes: (RoutesFn, RoutesFn, RoutesFn, RoutesFn, RoutesFn, RoutesFn)) -> Self {
+        VariadicRoutes {
+            _phantom_state: PhantomData,
+            routes: vec![routes.0, routes.1, routes.2, routes.3, routes.4, routes.5],
+        }
+    }
+}
```

Comment:
Could we only add variants for 1-3 tuples, then add an `impl From<Vec<RoutesFn>>` to cover 4+? (I tend to agree that any more than 3 versions is probably unlikely, but having `vec![]` as an escape hatch might be good.)

(An aside: I did something like this for storage backends in my git impl – implementing the behavior for `(LHS, RHS)` where both `LHS` and `RHS` individually implemented a "storage" trait, then implementing a terminator for `()`, so you could do `(a, (b, (c, d)))` or `()`; probably not as ergonomic in this case, but would allow for infinite length!)